### PR TITLE
Redesign tree page and add sections for site images and tree benefits

### LINKED ISSUE #240: Update My Trees Page
**Why**
- add in new features to the my trees page

**What to do**
- implement the tree images, environmental benefits, layouts etc.
- follow the below design spec:
> https://www.figma.com/proto/Rj1fh2FeRNrJlcVq161xcc/SFTT-2022-2023?page-id=206%3A410&node-id=416-812&viewport=-467%2C-1564%2C0.2&scaling=scale-down&starting-point-node-id=414%3A459&show-proto-sidebar=1

### PR DESCRIPTION
**Pull request by chromium-52**
## Checklist

- [ ] 1. Run `npm run check`
- [ ] 2. Run `npm run test`
- [ ] 3. Run `npm build --if-present`  

## Why

Resolves #240 

Redesigns the tree page according to the Figma designs
Adds two new sections on the tree page for the site image carousel and tree benefits

Note: This PR doesn't include adding/deleting site images from the frontend

## This PR

The most important changes are the two new components, `<SiteImageCarousel />` and `<TreeBenefits />`, as well as restructuring the tree page components (specifically, putting them in `src/components/treePage/`) with a [barrel file](https://basarat.gitbook.io/typescript/main-1/barrel) so that they're all in one place and not scattered throughout the `components` folder

## Screenshots

localhost_3000_tree_3303432

## Verification Steps

Visually checked the tree page for several trees (http://localhost:3000/tree/3303432, http://localhost:3000/tree/3175896, etc.)

### CODE REVIEW

File: src/containers/treePage/ducks/types.ts [hunk 1]
```diff
@@ -289,6 +290,34 @@ export interface AdoptedSites {
   adoptedSites: number[];
 }
 
+export interface SiteEntryImage {
+  imageId: number;
+  uploaderUsername: string | null;
+  uploadedAt: string;
+  imageUrl: string;
+}
```
Comment (huang0h) [inline src/containers/treePage/ducks/types.ts:298]: ⛏️ IIRC usernames are non-null, so I think we can type `uploaderUsername` as just `string`

**Review by huang0h — COMMENTED** [review]
💯 redesign looks awesome!! Thanks for working on this, I really should have split this over multiple tickets

File: src/components/treePage/siteImageCarousel.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    <>
+      {latestEntrySiteImages.length > 0 && (
+        <CarouselContainer>
+          <StyledCarousel
+            arrows
+            prevArrow={<LeftOutlined />}
+            nextArrow={<RightOutlined />}
+            adaptiveHeight
+            afterChange={onAfterChange}
+          >
```
[the remark this answers is not in the document]
Comment (chromium-52) [reply src/components/treePage/siteImageCarousel.tsx:74]: Hmm is [this](https://drive.google.com/file/d/1OnC3_HEGpyNdH8KmcDmjcLw13mqwxkC0/view?usp=sharing) what you're expecting to see if there are two images where one is uploaded by chromium52 and the other is by anonymous? I'm not 100% sure why but I can't seem to reproduce it on my end 🤔

[the remark this answers is not in the document]
Comment (huang0h) [reply src/components/treePage/siteImageCarousel.tsx]: Yeah, that's what should be happening - might be an issue on my end somewhere else then 🤔. Let's just merge this and see this issue happens in prod then

**Review by huang0h — APPROVED** [review]

File: src/components/treePage/siteImageCarousel.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (chromium-52) [reply src/components/treePage/siteImageCarousel.tsx]: Ok sounds good!

[the remark this answers is not in the document]
Comment (chromium-52) [reply src/components/treePage/siteImageCarousel.tsx]: Just verified this looks good in prod

https://github.com/Code-4-Community/speak-for-the-trees-frontend/assets/33704204/3bb042a3-d8d6-4d46-8f20-04ac5cdf7b03

**Pull request merged**